### ISSUE TITLE
julia: caching packages

### ISSUE THREAD

**Issue by jpfairbanks** [travis-ci/travis-ci#5744]
Is there a way to cache julia package dependencies?
For example using this technique to cache `$HOME/.julia/v0.4/Package.jl/`
https://docs.travis-ci.com/user/caching#Caching-directories

Has anyone successfully done this?

**Comment by BanzaiMan** [comment]
Paging @tkelman @ninjin @staticfloat @simonbyrne

**Comment by tkelman** [comment]
Yes you should be able to, but if you cache an existing version of a package in-place, it will go stale unless you call `Pkg.update()`. I also wouldn't recommend caching your own package this way (so if you want to try it, start with just dependencies), you want to do `Pkg.clone(pwd())` to make sure you're testing against either the branch or PR that Travis is running against.

I haven't tried caching an entire package myself, but I have done some caching of the compilation results of large C++ libraries that a package depends on. I can dig that up if that sounds like what you're trying to accomplish.

**Comment by staticfloat** [comment]
Is your desire to reduce network traffic through caching?  You can clone the repositories beforehand and cache those repos, but you will ned to update them on every test (as packages may need new versions of dependencies, etc...), so you cannot completely eliminate the network traffic.
